File: IKeep.Web/ClientApp/src/models/Building.js
import Entity from './Entity'

class Building extends Entity
{
    constructor(json)
    {
        super(json)
        if(json)
        {
            this.Ref = json.ref;
            this.Name = json.name;
            this.Description = json.description;
            this.InstallationId = json.installationId;
            this.FloorsIds = json.floorsIds;
        }
        else
        {
            this.Ref = "";
            this.Name = "";
            this.Description = "";
            this.InstallationId = "";
            this.FloorsIds = [];
        }
    }
}

export default Building;

File: IKeep.Lib/Models/Building.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class Building : Entity
    {
        public string Ref { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Guid InstallationId { get; set; }
        [JsonIgnore]
        public virtual ICollection<Floor> Floors { get; set; }
        public List<Guid> FloorsIds
        {
            get
            {
                return Floors == null ? new List<Guid>() : Floors.Select(x => x.Id).ToList();
            }
        }
    }
}